addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -58,6 +61,16 @@ def _compute_sms_has_insufficient_credit(self):
         for mailing in self:
             mailing.sms_has_insufficient_credit = mailing in mailing_ids
 
+    @api.depends('mailing_trace_ids.failure_type')
+    def _compute_sms_has_unregistered_account(self):
```

Comment:
Wouldn't be possible to make a single read_group, for both insufficient credit and unregistered account fields ? Reading mailing_id and failure_type, then grouping on both and if count > 0 then field is True. That was we do a single read_group for both fields and we avoid browsing a potentially large set of traces. What do you think ?